include/boost/math/distributions/cauchy.hpp

```diff
@@ -347,6 +347,13 @@ inline RealType kurtosis_excess(const cauchy_distribution<RealType, Policy>& /*d
       std::numeric_limits<RealType>::quiet_NaN(), Policy());
 }
 
+template <class RealType, class Policy>
+inline RealType entropy(const cauchy_distribution<RealType, Policy> & dist)
+{
+   using std::log;
+   return log(2*constants::two_pi<RealType>()*dist.scale());
+}
```

Comment:
I think integer `2` will implicitly convert to `double`, which could cause extra rounding when `RealType` is not `double`.
I would be strict and write it as `RealType(2)`.

Reply:
Not in any normal situation I can think of, it's actually a conceptual requirement of Boost.Math that all real-valued types interoperate with integer literals, and we make plenty of use of that throughout the library.  Besides I can think of no situation where `2.0` (double precision) looses bits compared to `2`.  The only situation I can think of is if RealType isn't constructible or multipliable by `int` and has a lossy _implicit_ conversion operator to double.
BTW in the multiprecision case, multiplying by an integer is likely to be significantly more efficient than constructing a temporary `RealType` and multiplying.

Reply:
> Not in any normal situation I can think of, it's actually a conceptual requirement of Boost.Math that all real-valued types interoperate with integer literals, and we make plenty of use of that throughout the library. Besides I can think of no situation where `2.0` (double precision) looses bits compared to `2`. The only situation I can think of is if RealType isn't constructible or multipliable by `int` and has a lossy _implicit_ conversion operator to double.

What I'm thinking of is the so-called ["double rounding"](https://www.exploringbinary.com/double-rounding-errors-in-floating-point-conversions/) error, when `RealType` is `float`, then the extra conversion from `double` back down to `float` incurs an extra rounding.

> BTW in the multiprecision case, multiplying by an integer is likely to be significantly more efficient than constructing a temporary `RealType` and multiplying.

Yes, I understand that Boost.Math probably has a policy of being performance-friendly to multi-precision types, but even at the cost of accuracy to other floating point types? I'm not expecting you to flip a long-standing policy right now, and I'm sure that C++ being the language it is could solve all situations to everyone's liking (but with a lot more work).

Surely this sort of small literal value performance penalty must arise frequently for multi-precision types? Could it not be solved in general with `constexpr` constructors or a small-object optimization?

Reply:
Doesn't look like the double rounding problem happens on gcc:

https://godbolt.org/z/m4q-vz

Reply:
Yes, it does look like clang and GCC optimize the integer literal away to a compile-time effect. Even integer non-literal values appeared to not have double rounding, so that's good.
I assumed that the literal `int` would promote to `double` first.

It's a different story when the literal is `2.0` and so actually has the `double` type. See for yourself, the `float` function now looks very inefficient. But that's not the actual situation here, so it's case closed for now. Something to think about though.  :)